src/cmap/auth/mongo_credentials.ts

```diff
@@ -62,7 +63,7 @@ export interface AuthMechanismProperties extends Document {
   /** A user provided OIDC human interacted callback function. */
   OIDC_HUMAN_CALLBACK?: OIDCCallbackFunction;
   /** The OIDC environment. Note that 'test' is for internal use only. */
-  ENVIRONMENT?: 'test' | 'azure' | 'gcp';
+  ENVIRONMENT?: 'test' | 'azure' | 'gcp' | 'k8s';
```

Comment:
(optional) I think you should be able to do:

```suggestion
  ENVIRONMENT?: (typeof ALLOWED_ENVIRONMENT_NAMES[number]);
```

If you want this to automatically update when we add new OIDC environments to ALLOWED_ENVIRONMENT_NAMES.

Reply:
I left as is for now as Typescript didn't like the suggestion and didn't want to fumble around with it too much as it's only 2 places in the codebase and seemed to me a bit more clear when seeing the actual strings in the properties themselves.